spec/controllers/hyrax/dashboard/collections_controller_spec.rb

```diff
@@ -82,6 +82,28 @@
       end
     end
 
+    context 'when setting collection type' do
+      let(:collection_type) { create(:collection_type) }
+
+      it "creates a Collection of default type when type is nil" do
+        expect do
+          post :create, params: {
+            collection: collection_attrs
+          }
+        end.to change { Collection.count }.by(1)
+        expect(assigns[:collection].collection_type.machine_id).to eq Hyrax::CollectionType::DEFAULT_ID
+      end
+
+      it "creates a Collection of specified type" do
+        expect do
+          post :create, params: {
+            collection: collection_attrs.merge(collection_type_gid: collection_type.gid)
+          }
+        end.to change { Collection.count }.by(1)
+        expect(assigns[:collection].collection_type_gid).to eq collection_type.gid
```

Comment:
In the previous test, I am specifically checking that the special 'user_collection' was created.  If I check the gid, it could be that the factory happened to create the collection_type with gid and I would never know it.  Where as in this second test, the factory should be creating a new type with gid.  I just need them to be the same.